### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by nraanes**

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+    implementation group: 'commons-io', name: 'commons-io', version: '2.6'
+    implementation group: 'org.apache.commons', name: 'commons-lang3', version: '3.9'
+    implementation group: 'javax.ws.rs', name: 'javax.ws.rs-api', version: '2.1.1'
+    implementation group: 'com.google.code.gson', name: 'gson', version: '2.8.6'
+    implementation group: 'com.amazonaws', name: 'aws-lambda-java-core', version: '1.2.0'
+    implementation group: 'com.amazonaws.secretsmanager', name: 'aws-secretsmanager-caching-java', version: '1.0.1'
+    testImplementation group: 'org.junit.jupiter', name: 'junit-jupiter-engine', version: '5.5.2'
+    testImplementation group: 'org.mockito', name: 'mockito-core', version: '2.1.0'
+
+    shadowJar {
```
Comment (janbj) [inline build.gradle:30]: Do we need shadowJar for Lambda?

File: build.gradle [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+jacocoTestReport {
+    reports {
+        xml.enabled false
+        csv.enabled false
+        html.destination file("${buildDir}/reports/jacoco")
+    }
+}
+
+jacoco {
+    toolVersion = "0.8.5"
```
Comment (janbj) [inline build.gradle:77]: We do not need to specify version, Gradle already handles it

File: buildspec.yml [hunk 3]
```diff
@@ -0,0 +1,23 @@
+version: 0.2
+
+phases:
+
+  install:
+    runtime-versions:
+      java: openjdk8
```
Comment (janbj) [inline buildspec.yml:7]: openjdk11

File: src/main/java/no/unit/nva/datacite/Config.java [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+    public static final String ERROR_MISSING_ENVIRONMENT_VARIABLES = "Missing environment variables";
+
+    public static final String ENVIRONMENT_NAME_CORS_ALLOW_ORIGIN_HEADER = "ALLOWED_ORIGIN";
+    public static final String ENVIRONMENT_NAME_DATACITE_MDS_CONFIGS = "DATACITE_MDS_CONFIGS";
+
+    private String corsHeader;
+    private String dataCiteMdsConfigs;
+
+
+    private Config() {
```
Comment (janbj) [inline src/main/java/no/unit/nva/datacite/Config.java:16]: You should add nva-commons as a dependency with Jitpack: https://github.com/BIBSYSDEV/nva-commons

This is our new micro-framework for common code such as reading environment variables.

File: src/main/java/no/unit/nva/datacite/DataCiteMdsCreateDoiHandler.java [hunk 5]
```diff
@@ -0,0 +1,226 @@
...
+import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.ConcurrentHashMap;
+
+
+
+/**
+ * Handler for requests to Lambda function.
+ */
+public class DataCiteMdsCreateDoiHandler implements RequestHandler<Map<String, Object>, GatewayResponse> {
```
Comment (janbj) [inline src/main/java/no/unit/nva/datacite/DataCiteMdsCreateDoiHandler.java:26]: Use nva-commons?

Comment (nraanes) [reply src/main/java/no/unit/nva/datacite/DataCiteMdsCreateDoiHandler.java:26]: I made PR on wrong branch :P

**Pull request closed without merging**